### PR TITLE
Retry when request to whois ip fail.

### PR DESCRIPTION
**Pull request by mohsinenar**

### CODE REVIEW

**Review by PiranhaSa — APPROVED** [review]
I think you need to use black

File: tests/whois_ip_agent_test.py [hunk 1]
```diff
@@ -205,3 +205,22 @@ def testAgentWhoisIP_whenDomainScopeArgAndDnsRecordMsgNotInScope_targetShouldNot
     whois_ip_agent_with_scope_arg.process(msg)
 
     assert len(agent_mock) == 0
+
+
+def testAgentWhoisIP_whenRDAPIsDown_shouldRetry(
+    mocker,
```
Comment (deadly-panda) [inline tests/whois_ip_agent_test.py:211]: Add the type hint please.

File: agent/whois_ip_agent.py [hunk 2]
```diff
@@ -80,12 +94,15 @@ def _process_dns_record(self, message: m.Message) -> None:
             if self.set_add("agent_whois_ip_asset", host):
                 logger.info("processing ip %s", host)
                 try:
-                    record = ipwhois.IPWhois(host).lookup_rdap()
+                    record = _get_wohis_record(host)
```
Comment (BlueSquare1) [inline agent/whois_ip_agent.py:97]:
```suggestion
                    record = _get_whois_record(host)
```

**Review by BlueSquare1 — DISMISSED** [review]

**Review by elyousfi5 — APPROVED** [review]

File: agent/whois_ip_agent.py [hunk 3]
```diff
@@ -24,6 +25,19 @@
 )
 logger = logging.getLogger(__name__)
 
+MAX_RETRY_ATTEMPTS = 5
+WAIT_BETWEEN_RETRY = 3
```
Comment (3asm) [inline agent/whois_ip_agent.py:29]:
```suggestion
MAX_RETRY_ATTEMPTS = 2
WAIT_BETWEEN_RETRY = 3
```

File: agent/whois_ip_agent.py [hunk 4]
```diff
@@ -80,12 +94,15 @@ def _process_dns_record(self, message: m.Message) -> None:
             if self.set_add("agent_whois_ip_asset", host):
                 logger.info("processing ip %s", host)
                 try:
-                    record = ipwhois.IPWhois(host).lookup_rdap()
+                    record = _get_whois_record(host)
```
[the remark this answers is not in the document]
Comment (3asm) [reply agent/whois_ip_agent.py:97]: +1

File: tests/whois_ip_agent_test.py [hunk 5]
```diff
@@ -205,3 +205,22 @@ def testAgentWhoisIP_whenDomainScopeArgAndDnsRecordMsgNotInScope_targetShouldNot
...
+    """Test collecting whois of an IPv4 address, when server is down should retry."""
+    del agent_persist_mock
+    mock_request = mocker.patch(
+        "urllib.request.OpenerDirector.open", return_result=mocker.Mock(status=501)
+    )
+
+    test_agent.process(scan_message_ipv4_mask)
+
+    assert len(agent_mock) == 0
+    assert mock_request.call_count == 10
```
[the remark this answers is not in the document]
Comment (3asm) [reply tests/whois_ip_agent_test.py:227]: Make the assert explicit with 2 * len(scan_message_ipv4_mask)

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**